src/backend/distributed/connection/placement_connection.c

```diff
@@ -1075,6 +1075,17 @@ CheckShardPlacements(ConnectionShardHashEntry *shardEntry)
 			if (shardPlacement->shardState == FILE_FINALIZED)
 			{
 				UpdateShardPlacementState(placementEntry->key.placementId, FILE_INACTIVE);
```

Comment:
Could we move this into a separate function? e.g. `MarkShardPlacementInactive(..)` which also implements the partitioning-specific logic?